### PR TITLE
(chore) move moonscript into separate C file

### PR DESCRIPTION
**Pull request by joshgoebel**
Feels like every language should have their own C file in `api`... so I wonder if this should be broken out?

### CODE REVIEW

**Comment by joshgoebel** [comment]
Would this be good?  If so I can do the same with Fennell...

File: src/api/squirrel.c [hunk 1]
```diff
@@ -1447,7 +1447,7 @@ static void squirrel_open_builtins(HSQUIRRELVM vm)
     sq_poptop(vm);
 }
 
-static void initAPI(tic_core* core)
+static void initLuaAPI(tic_core* core)
```
Comment (nesbox) [inline src/api/squirrel.c:1450]: It looks like the function was accidentally renamed here, there should be no mention of Lua in the Squirrel module.

File: src/api/wren.c [hunk 2]
```diff
@@ -1395,7 +1395,7 @@ static WrenForeignMethodFn bindForeignMethod(
     return foreignTicMethods(fullName);
 }
 
-static void initAPI(tic_core* core)
+static void initLuaAPI(tic_core* core)
```
Comment (nesbox) [inline src/api/wren.c:1398]: Same here, please fix if not difficult.
Thank you.

**Review by nesbox — CHANGES_REQUESTED** [review]

**Comment by joshgoebel** [comment]
Yeah, no problem.  I'll fix this up and make sure the tests are green.  Just didn't want to throw more work at this if you aren't ok with the change in general.

**Review by nesbox — APPROVED** [review]
thank you

**Comment by joshgoebel** [comment]
Yeah, not sure this was 100% ready yet. :)  You got to it too fast.

**Comment by nesbox** [comment]
ok, waiting... :)

**Comment by joshgoebel** [comment]
But you already merged it... did you get it working or should I just open a new PR? :-)

**Comment by nesbox** [comment]
Honestly, I don't know what to do in this case :)

**Pull request merged**